client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -1531,6 +1548,12 @@ public void onSuccess(ByteBuffer response) {
                 pushState.onSuccess(hostPort);
                 callback.onSuccess(ByteBuffer.wrap(new byte[] {StatusCode.SOFT_SPLIT.getValue()}));
               } else {
+                if (dataPushFailureTrackingEnabled) {
+                  for (int i = 0; i < numBatches; i++) {
```

Comment:
Only need add batchesNeedResubmit to FailedBatch~

Reply:
Thanks a lot, will fix and push later